analysis/freq.go

```diff
@@ -55,28 +55,29 @@ func (tfs TokenFrequencies) MergeAll(remoteField string, other TokenFrequencies)
 func TokenFrequency(tokens TokenStream, arrayPositions []uint64) TokenFrequencies {
 	rv := make(map[string]*TokenFreq, len(tokens))
 
+	tls := make([]TokenLocation, len(tokens))
+	tlNext := 0
+
 	for _, token := range tokens {
-		curr, ok := rv[string(token.Term)]
+		tls[tlNext] = TokenLocation{
+			ArrayPositions: arrayPositions,
+			Start:          token.Start,
+			End:            token.End,
+			Position:       token.Position,
+		}
+
+		term := string(token.Term)
+		curr, ok := rv[term]
```

Comment:
Is this for optimizing allocations and if so did you actually check the allocation count of this before and after? When writing `rv[string(token.Term)]` the compiler will avoid allocating a new string since Go 1.3 because it knows that the string is only used once for hashing and it won't be modified. But if you split the conversion and the map access (like you did here) the compiler can not optimize this  anymore (since it can't guarantee that the underlying bytes won't be modified between the conversion and the map access) and will need to allocate the string. I haven't compiled the code so it could be that the string still won't be allocated on the heap, but from my naive guess, it likely will need an allocation.